### PR TITLE
return 204 when subset is empty

### PR DESCRIPTION
**Pull request by dmgaldi**
## Overview
Updating logic in data service to return a 204 status code instead of 500 when a tabular subset is empty.

## Testing:
Ran the following request and validated I got back a 204:

```
{
    "studyId": "VBP_MEGA",
    "filters": [
        {
            "type": "dateRange",
            "entityId": "OBI_0000659",
            "variableId": "EUPATH_0043256",
            "min": "1928-03-18T00:00:00Z",
            "max": "1956-03-20T00:00:00Z"
        }
    ],
    "config": {
        "geoAggregateVariable": {
            "entityId": "GAZ_00000448",
            "variableId": "EUPATH_0043203"
        },
        "latitudeVariable": {
            "entityId": "GAZ_00000448",
            "variableId": "OBI_0001620"
        },
        "longitudeVariable": {
            "entityId": "GAZ_00000448",
            "variableId": "OBI_0001621"
        },
        "overlayConfig": {
            "overlayType": "categorical",
            "overlayVariable": {
                "variableId": "OBI_0001909",
                "entityId": "EUPATH_0000609"
            },
            "overlayValues": [
                "Aedes vexans",
                "Culex pipiens morphological group",
                "Anopheles <subgenus>",
                "Culex tarsalis",
                "Culex nigripalpus",
                "Anopheles punctipennis",
                "Culex erraticus",
                "__UNSELECTED__"
            ]
        },
        "outputEntityId": "EUPATH_0000609",
        "valueSpec": "count",
        "viewport": {
            "latitude": {
                "xMin": -45.08903556483102,
                "xMax": 77.07878389624943
            },
            "longitude": {
                "left": -170.15624999,
                "right": -170.15625
            }
        }
    }
}
```

### CODE REVIEW

**Review by ryanrdoherty — APPROVED** [review]

**Comment by dmfalke** [comment]
I made two issues for the web client that need to be merged in coordination with this:
- https://github.com/VEuPathDB/web-monorepo/issues/606
- https://github.com/VEuPathDB/web-monorepo/issues/607

**Comment by dmfalke** [comment]
I did a quick test with this, using the first reproduction case in https://github.com/VEuPathDB/EdaDataService/issues/312#issue-1865872608.

If we don't make any changes on the client, the UX is identical (the details that appear when clicking "Show more" are different):
image

I'd like to propose we wait until after build 66 is released to merge this, primarily to reduce the number of changes needed on the client.

@bobular do you agree?

File: src/main/java/org/veupathdb/service/eda/ds/core/AbstractPlugin.java
Comment (ryanrdoherty) [inline src/main/java/org/veupathdb/service/eda/ds/core/AbstractPlugin.java:182]: It's funny how this makes me very nervous not to have this just sitting "naked" between creation here and the calling of the Consumer<OutputStream>.  But should be safe in theory.

**Review by ryanrdoherty — APPROVED** [review]

**Pull request merged**
